### PR TITLE
Fethas Carnival of Horrors part 4: I am in your greytide guarding ur newbs

### PR DESCRIPTION
**Pull request by Aurorablade**
Ports guardians from TG (i cannot for the life of me find the orignal PR for this....)

Guardians are entities such as spirits that inhabit a person and help them out.

They can manifest or recall from a person, but if they get 10 meters away they are pulled back (need to fix this in the code so they demanifest...)
They are technically invincible but any damage they DO take is transferred to their user, and if the user dies, they die.

Currently commented out:dusting the user when they die...i can put it back in, but i am unsure of it so commented out.

Guardian Themes:
Magical(Stand/Spirit): Summoned from a deck of tarot cards. CURRENTLY NOT IN GAME! I WOULD LOVE SUGGESTIONS ON HOW THIS SHOULD BE SPAWNED AND BY WHAT!

Tech(Holoparasite):Summoned by hypo injection. Bought from traitor uplink.

Bio(Scarab swarm, currently represented by a worm thing):Summoned by egg cluster. CURRENTLY NOT IN GAME! I WOULD LOVE SUGGESTIONS ON HOW THIS SHOULD BE SPAWNED AND BY WHAT!

Types of Guardians:

Chaos: As a Chaos type, you have only light damage resistance, but will ignite any enemy you bump into. In addition, your melee attacks will randomly teleport enemies.
Ignites mobs on contact. Teleports enemies away. Will not ignite user.

Standard:As a standard type you have no special abilities, but have a high damage resistance and a powerful attack capable of smashing through walls.
You punch things..alot of things..you say a battlecry when you punch things (getready for HONKHONKHONKHONK)

Support:
As a Support type, you may toggle your basic attacks to a healing mode. In addition, Alt-Clicking on an adjacent mob will warp them to your bluespace beacon after a short delay.

Two modes, healer and combat. combat is low damage. Healing mode will heal a thing on attack by 5.
Can set up a beacon and warp mobs to it.

Ranged:
As a ranged type, you have only light damage resistance, but are capable of spraying shards of crystal at incredibly high speed. You can also deploy surveillance snares to monitor enemy movement. Finally, you can switch to scout mode, in which you can't attack, but can move without limit.

BOMB:
As an explosive type, you have only moderate close combat abilities, but are capable of converting any adjacent item into a disguised bomb via alt click.

These are chosen at random on someone using the creator item. There are choosable variants that are admin spawnable.

### CODE REVIEW

**Comment by ZomgPonies** [comment]
Interesting for events, seems self-contained, :+1:

**Comment by Mijhead** [comment]
yup _efd9e8dcac01dd8433fca5412ac559c8

**Comment by mrsdobbins** [comment]
http://i.imgur.com/fui2fTi.gif
SEKAI ICHI.

**Comment by Spacemanspark** [comment]
:+1:

**Comment by Ultimate-Chimera** [comment]
:+1:

**Comment by Fox-McCloud** [comment]
I was going to port these, because they're pretty neat, but never got around to it.

Should be interesting.

**Comment by taukausanake** [comment]
I thought people didn't like this because it was OP? I really like this but have only one feature I'd like to add. For a Revenant event give a random chance for a Guardian to be spawned/granted/adminbused and give the Guardian the objective to kill the Revenant. Spirit against spirit. It would give the chaplain a break for a while

**Comment by Aurorablade** [comment]
eh its an odd issue on the opness, the solution there was to have the summoner be dusted when they died (IE Uncloneable)...i am unsure on that but will keep it in if its the only way (IE raise a TC cost) i also would like feedback on where the Bio and Terot ones can be gained.

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -142,7 +142,12 @@ proc/isembryo(A)
 		return 1
 	return 0
 
-proc/isobserver(A)
```
Comment (Fox-McCloud) [inline code/modules/mob/mob_helpers.dm]: github, why you do this.

**Comment by Fox-McCloud** [comment]
>These are chosen at random on someone using the creator item. There are choosable variants that are admin spawnable.

This is incorrect; the one from the traitor uplink allows you to select which one it will be when you inject it (thus why it's a subtype called pick with the random var set to false).

IMO, a good thing.

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (Aurorablade) [inline code/modules/mob/mob_helpers.dm]: OH YOU ADDED SOMETHING BETWEEN TWO LINES?! HERE LET ME JUST SHOW YOU DELETED THE WHOLE THING!

**Comment by Aurorablade** [comment]
oh i missed that...

**Comment by Mijhead** [comment]
<BlackCat> I feel like Stand users should spawn in vampire rounds.
<Mrs_Dobbins> Nah
<Mrs_Dobbins> Actually thats a cool idea
<Mrs_Dobbins> b-but they'd need some objective related to the vampires
<BlackCat> Though that'd probably require Vampires to get better or also have stands.
<Mrs_Dobbins> Just make the vampires more like the pillar men

;^)

**Comment by mrsdobbins** [comment]
B-but shadowlings are, to all intents and purposes just pillar men...

**Comment by Aurorablade** [comment]
@mijhead ...That gives me an idea..but i would need to limit it to like one or two types (We do NOT need vampires running around with Bomb and choas sprits) and i still need to figure out where to spawn Bio and Magic...Magic COULD go to wizard...MAYBE go to wizard....

**Comment by Mijhead** [comment]
What about as an evolution of some sorts of borers?

**Comment by Dumbdumn5** [comment]
:+1: X :100:
Now I can finally have a way to say that I'm somebody's guardian angel without being a small card or a seal that does nothing but make rude comments.

**Comment by Earthdivine** [comment]
Recommend giving Magic to the Wizard, he could stand having a second man to watch over him for Scientists with sprays.

**Comment by Aurorablade** [comment]
yeah will do that. Also for vampires..i was considering a bat, as a power, limited to healer,standard and ranged/scout type. What i need to figure out is..make the power unuseable while a bat is bound to you...and thinking of if it would overlap with the Summon Bat power....

.....Also maybe i should make a admin only seal version for @Dumbdumn5

**Comment by Aurorablade** [comment]
not gonna do the vampire thing..AM going to do the seal thing, i need someone to tell me how to make it truely adminbussable here so it dosen't pick a random player for it

**Comment by Aurorablade** [comment]
i also just realized i forgot to give the magic creator to the wizard in last commit...i knew i was forgetting something...

**Comment by Aurorablade** [comment]
Changed Seal on Feedback, Fixed Intent so it stops nuzzling things..hopefully....
Added wizard terot to spellbook

Help needed on:
Admin Seal needs to not pick a random player, but the admin who uses it and attach to a player.
Where do i put the Bio themed ones?

**Comment by Aurorablade** [comment]
Added bio guardian spawner to Xenden asteroid treasure
modified battlecry, I need someone to CHECK the code here mainly cuase its not triggering when i a spawn a mob to punch, though that was before i did one change.

LAST THING TO DO:
Make the seal not choose a random player but the admin attempting to use it...no IDEA WHAT I AM DOING.

**Comment by Earthdivine** [comment]
Throwing this here since I said it in the super busy chat, you probably will have to have it check for players with R_ADMIN or something. Forgot what it is.

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (Tastyfish) [inline code/modules/mob/mob_helpers.dm]: The line's different from the original one, you silly people.

**Comment by Aurorablade** [comment]
Tried a swarmer fic seems to work (when i move away from what i dismantle..it does stop the process)
Adminbuss seal needs a double check on the code.
Otherwise done.
Review plz

**Comment by Aurorablade** [comment]
well i mean review when i can commit this....*watchs git for windows

File: code/game/asteroid.dm [hunk 2]
```diff
@@ -97,7 +97,7 @@ var/global/max_secret_rooms = 6
 			theme = "xenoden"
 			walltypes = list(/turf/simulated/mineral/random/high_chance=1)
 			floortypes = list(/turf/simulated/floor/plating/airless/asteroid, /turf/simulated/floor/beach/sand)
-			treasureitems = list(/obj/item/clothing/mask/facehugger=1,/obj/item/stack/sheet/animalhide/xeno=2,/obj/item/clothing/suit/xenos=2,/obj/item/clothing/head/xenos=2)
+			treasureitems = list(/obj/item/clothing/mask/facehugger=1,/obj/item/stack/sheet/animalhide/xeno=2,/obj/item/clothing/suit/xenos=2,/obj/item/clothing/head/xenos=2,/obj/item/weapon/guardiancreator/biological=4)
```
Comment (TheDZD) [inline code/game/asteroid.dm]: The chance for a fairly powerful item probably shouldn't be that high, might be best to set its weight to 1.

**Comment by TheDZD** [comment]
Besides that, I didn't find much to point out. Looks good otherwise.

**Comment by Aurorablade** [comment]
okay commiting the treasure room tweak BUT can anyone see a reason WHY the battle cry is not triggering on attacking a mob?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -0,0 +1,895 @@
...
+				src << "<span class='danger'><B>You need a beacon to warp things!</span></B>"
+	else
+		src << "<span class='danger'><B>You need to hold still!</span></B>"
+
+
+///////////////////Ranged
+
+/obj/item/projectile/guardian
+	name = "crystal spray"
+	icon_state = "guardian"
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: You never included this icon; it'll be invisible.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 4]
```diff
@@ -0,0 +1,895 @@
...
+	icon_living = "stand"
+	speed = 0
+	a_intent = I_HARM
+	stop_automated_movement = 1
+	floating = 1
+	attack_sound = 'sound/weapons/punch1.ogg'
+	minbodytemp = 0
+	maxbodytemp = 10000000
+	min_oxy = 0
+	max_oxy = INFINITY
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Shouldn't this be 0?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 5]
```diff
@@ -0,0 +1,895 @@
...
+	min_tox = 0
+	max_tox = INFINITY
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: as above

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 6]
```diff
@@ -0,0 +1,895 @@
...
+	min_co2 = 0
+	max_co2 = INFINITY
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: as above

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 7]
```diff
@@ -0,0 +1,895 @@
...
+	min_n2  = 0
+	max_n2  = INFINITY
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: as above

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 8]
```diff
@@ -0,0 +1,895 @@
...
+	attacktext = "punches"
+	maxHealth = 100000 //The spirit itself is invincible
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Should be infinity.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 9]
```diff
@@ -0,0 +1,895 @@
...
+	health = 100000
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Should be infinity.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 10]
```diff
@@ -0,0 +1,895 @@
...
+	var/animated_manifest = FALSE
+	var/cooldown = 0
+	var/damage_transfer = 1 //how much damage from each attack we transfer to the owner
+	var/mob/living/summoner
+	var/range = 10 //how far from the user the spirit can be
+	var/playstyle_string = "You are a standard Guardian. You shouldn't exist!"
+	var/magic_fluff_string = " You draw the Coder, symbolizing bugs and errors. This shouldn't happen! Submit a bug report!"
+	var/tech_fluff_string = "BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!"
+	var/bio_fluff_string = "Your scarabs fail to mutate. This shouldn't happen! Submit a bug report!"
+	var/admin_fluff_string = "URK URF!"//the wheels on the bus...
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Should probably add the butcher result of this as being ectoplasm, as well; though it's by no means high priority.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 11]
```diff
@@ -0,0 +1,895 @@
...
+		if(summoner.stat == DEAD)
+		//	src << "<span class='danger'>Your summoner has died!</span>"
+		//	visible_message("<span class='danger'><B>The [src] dies along with its user!</B></span>")
+		//	summoner.visible_message("<span class='danger'><B>[summoner]'s body is completely consumed by the strain of sustaining [src]!</B></span>")
+		//	for(var/obj/item/W in summoner)
+			//	if(!summoner.unEquip(W))
+			//		qdel(W)
+			//summoner.dust()
+			//ghostize()
+			//qdel(src)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: I realize you wanted to get rid of the um dust() thing, but shouldn't it still delete the parasite? Just checking.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 12]
```diff
@@ -0,0 +1,895 @@
...
+		if (2)
+			adjustBruteLoss(60)
+
+		if(3)
+			adjustBruteLoss(30)
+
+/mob/living/simple_animal/hostile/guardian/gib()
+	if(summoner)
+		summoner << "<span class='danger'><B>Your [src] was blown up!</span></B>"
+		summoner.gib()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Just wanted to make sure you're ok with including this; you didn't want Dust() included--is this ok?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 13]
```diff
@@ -0,0 +1,895 @@
...
+	ghostize()
+	qdel(src)
+
+/mob/living/simple_animal/hostile/guardian/start_pulling(var/atom/movable/AM)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Don't see the big deal about them being able to pull things.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 14]
```diff
@@ -0,0 +1,895 @@
...
+			pickedtype = /mob/living/simple_animal/hostile/guardian/bomb
+
+	var/mob/living/simple_animal/hostile/guardian/G = new pickedtype(user)
+	G.summoner = user
+	G.key = key
+	G << "You are a [mob_name] bound to serve [user.real_name]."
+	G << "You are capable of manifesting or recalling to your master with verbs in the Guardian tab. You will also find a verb to communicate with them privately there."
+	G << "While personally invincible, you will die if [user.real_name] does, and any damage dealt to you will have a portion passed on to them as you feed upon them to sustain yourself."
+	G << "[G.playstyle_string]"
+	user.verbs += /mob/living/proc/guardian_comm
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: forgot guardian recall and guardian reset

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 4 above)
Comment (tigercat2000) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: If it was 0, the guardians would take damage any time they breath in oxygen.

**Pull request merged**